apps/extension/src/background/keyring/keyring.ts

```diff
@@ -138,6 +143,48 @@ export class KeyRing {
     return false;
   }
 
+  // Reset password by re-encrypting secrets with new password
```

Comment:
This works very well for me the first time I use it! I did notice something though: If I change my password, it doesn't always force me to re-enter it when I go to derive a new account (unless maybe the service worker stops)? So, if I go to reset password, and immediately go to derive a new address, I get an error if it doesn't prompt me for the new password. There's a couple things we might do: Either 1) explicitly lock the keyring after resetting password, or maybe better, 2)  Set the password in Keyring to the new password here. Otherwise, this all looks good!

Reply:
Ah! Thanks for spotting that. I've updated it now to change `this._password` on reset.